### PR TITLE
Fixed order dependency and removed no-shuffle-tag in build_ios_framew…

### PR DESCRIPTION
**Pull request by Swiftaxe**
Fixed order dependency and removed no-shuffle-tag in build_ios_framework_test.dart. Part of #85160.

Another one of these down. This one was a tad more tricky.

## The Problem
The tests were failing when using --test-randomize-ordering-seed=1000, and most other seeds as well.

The first three tests in this file set the Cache.flutterRoot variable. This is done by default by the testUsingContext method.
When randomizing the test order the method setUp on line 143 runs cache.getLicenseFile() without Cache.flutterRoot being set. The method getLicenseFile is dependent on flutterRoot being set, so this makes all test in that group fail.

## The Solution
Set the Cache.flutterRoot variable, if not set, before calling the method cache.getLicenseFile().

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by skia-gold** [comment]
Gold has detected about 1 new digest(s) on patchset 3.
View them at https://flutter-gold.skia.org/cl/github/94699

File: packages/flutter_tools/test/commands.shard/hermetic/build_ios_framework_test.dart [hunk 1]
```diff
@@ -62,6 +56,7 @@ void main() {
           fileSystem: memoryFileSystem,
           processManager: FakeProcessManager.any(),
         );
+
```
Comment (Piinks) [inline packages/flutter_tools/test/commands.shard/hermetic/build_ios_framework_test.dart:59]: Nit: revert

File: packages/flutter_tools/test/commands.shard/hermetic/build_ios_framework_test.dart [hunk 2]
```diff
@@ -146,6 +141,8 @@ void main() {
         const String licenseText = 'This is the license!';
 
         setUp(() {
+          // cache.getLicenseFile() relies on the flutter root being set.
```
Comment (Piinks) [inline packages/flutter_tools/test/commands.shard/hermetic/build_ios_framework_test.dart:144]: Nice find!

**Review by Piinks — COMMENTED** [review]
Can you rebase this to resolve the current test failures? Thanks!

File: packages/flutter_tools/test/commands.shard/hermetic/build_ios_framework_test.dart (hunk 1 above)
Comment (Swiftaxe) [reply packages/flutter_tools/test/commands.shard/hermetic/build_ios_framework_test.dart:59]: > Can you rebase this to resolve the current test failures? Thanks!

Sorry, you lost me 🙈.  Do you want me to rebase, or revert and what commit are we talking about? 
Can you be more specific please? 😊

Comment (Piinks) [reply packages/flutter_tools/test/commands.shard/hermetic/build_ios_framework_test.dart:59]: Oh sorry to be confusing. I meant to ask to revert any unnecessary formatting changes, and to rebase your PR with the latest from master.
The failing `customer_testing` shard tends to have issues as a PR gets older. Merging in updates from master don't work to resolve it when this happens, it requires rebasing. I hope this is a bit clearer. Let me know if it is not. :)

Comment (Swiftaxe) [reply packages/flutter_tools/test/commands.shard/hermetic/build_ios_framework_test.dart]: Thank you! Now I get it =)

**Review by christopherfujino — APPROVED** [review]
LGTM

**Pull request merged**